src/reconcile/rec_write.c

```diff
@@ -2753,6 +2763,24 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
     __wt_verbose(session, WT_VERB_RECONCILE, "%p reconciled into %" PRIu32 " pages", (void *)ref,
       r->multi_next);
 
+    /*
+     * TODO: This is still not the correct place. We may still fail after this. Let's put it here
+     * for now to minimize code changes. We should decide where it should be. It's not clear how we
+     * will do error handling in disaggreated storage. What should we do if we only write a subset
+     * of split pages successfully? Can we discard the pages we already written and start again?
```

Comment:
I bet we can discard, they will never be part of a checkpoint, so they should age out, right?

Reply:
Yes, I think so, in the on-disk version of wiredtiger the blocks should be freed eventually as they are not in any checkpoint. But I'm not fully sure in the case for disaggregated storage.

Another problem is that after we have marked the updates durable, we may still fail in the following code. This will confuse us to think the write fails and discard what we have written. Then if we retry reconciliation again, we will get the wrong results as the information on the update chain tells us that the previous reconciliation was successful.

I tried to make this the last step in the reconciliation but the code flow is a bit hard to achieve that. Therefore, I put it here temporarily.